src/main/java/microsoft/exchange/webservices/data/core/request/ServiceRequestBase.java

```diff
@@ -396,6 +396,39 @@ private static InputStream getResponseErrorStream(HttpWebRequest request)
   }
 
   /**
+   * Reads the response stream.
+   *
+   * @return serviceResponse
+   * @throws Exception
+   */
+  protected InputStream readResponseStream(HttpWebRequest response) throws Exception {
+    T serviceResponse;
+
+    if (!response.getResponseContentType().startsWith("text/xml")) {
+      String line = new BufferedReader(new InputStreamReader(ServiceRequestBase.getResponseStream(response)))
+          .readLine();
+      throw new ServiceRequestException("The response received from the service didn't contain valid XML.");
+    }
+
+    /**
+     * If tracing is enabled, we read the entire response into a
+     * MemoryStream so that we can pass it along to the ITraceListener. Then
+     * we parse the response from the MemoryStream.
+     */
+    try {
+      this.getService().processHttpResponseHeaders(TraceFlags.EwsResponseHttpHeaders, response);
```

Comment:
Comment like:
// Believe we need to call this before getting the response stream